src/main/kotlin/org/codefreak/codefreak/service/file/FileCollectionRepositoryService.kt

```diff
@@ -0,0 +1,238 @@
+package org.codefreak.codefreak.service.file
+
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.UUID
+import org.apache.tomcat.util.http.fileupload.FileUtils
+import org.codefreak.codefreak.config.AppConfiguration
+import org.codefreak.codefreak.service.AnswerService
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.api.InitCommand
+import org.eclipse.jgit.api.errors.CanceledException
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.Ref
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevWalk
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+
+@Service
+class FileCollectionRepositoryService(val config: AppConfiguration, val answerService: AnswerService) : IFileCollectionRepository {
+
+  override fun saveChanges(collectionID: UUID, commitMessage: String, force: Boolean): Boolean {
+    val workingTreePath = createCollectionPath(collectionID)
+    val gitDir = getGitDir(collectionID)
+    if (!Files.exists(Paths.get(gitDir))) {
+      val git = InitCommand()
+        .setDirectory(workingTreePath.toFile())
+        .setGitDir(File(gitDir))
+        .call()
+      git.add().addFilepattern(".").call()
+      git.commit().setMessage(commitMessage).call()
+      setCurrentHeadCommitId(git, collectionID)
+      return true
+    } else {
+      // fetch git repository from .git in folder
+      val git = Git.open(File(gitDir))
+      if (gotTouched(git) || force == true) {
+        val missingFiles = git.status().call().missing
+        for (remove in missingFiles) {
+          git.rm().addFilepattern(remove).call()
+        }
+        val addedFiles = git.status().call().added
+        for (add in addedFiles) {
+          git.add().addFilepattern(add).call()
+        }
+        git.add().addFilepattern(".").call()
+        git.commit().setMessage(commitMessage).call()
+        setCurrentHeadCommitId(git, collectionID)
+        return true
+      }
+      return false
+    }
+  }
+
+  /**
+   * Function which checks if any changes where made to the version.
+   * @param git the Jgit git instance to operate on
+   * @return true if any changes where made
+   */
+  private fun gotTouched(git: Git): Boolean {
+    // source: https://github.com/centic9/jgit-cookbook/blob/master/src/main/java/org/dstadler/jgit/porcelain/ListUncommittedChanges.java
+    val status = git.status().call()
+
+    val added = status.added
```

Comment:
nitpick (non-blocking): all the local variables are not really necessary. Also, this would be be much more concise as `return added.isNotEmpty() || changed.isNotEmpty() || ...`

Reply:
I aggree. Both works but I preferred this one for clarity/readability